snippets/card-product.liquid

```diff
@@ -384,7 +384,31 @@
             {%- endif -%}
           </div>
         {% elsif quick_add == 'bulk' %}
-          {% if card_product.variants.size == 1 %}
+          {% if card_product.selected_or_first_available_variant.available == false %}
```

Comment:
It looks like we have some extra logic here for buy buttons for when it's supposed to say sold out

https://github.com/Shopify/dawn/blob/main/snippets/buy-buttons.liquid#L77-L79

Reply:
ooh that's interesting. I copied this condition from the logic for quick add standard: https://github.com/Shopify/dawn/pull/3487/files#diff-167fc70beb2d8c5dbf3e81a213fd3808f6b2e1d1c60590a6863724dc0dca3ea6R368-R372

Do we want to keep this change consistent with quick add standard, or update both to copy what we're doing for buy-buttons? I'm not sure why the buy-button condition and quick add standard are currently different, is there risk in changing how we do it for quick add standard?

Reply:
@ludoboludo do you have context on this?

Reply:
Nope. I don't think there is a specific reason. My guess would be that when we added quantity rule logic to the theme we didn't apply to the quick add logic 🤷 Seems like it could have been an easy miss

Reply:
I could see how the logic for quantity rules could be useful. I haven't tested it though 🤔 
The rest of logic with `inventory_management` and `inventory_policy` isn't needed because it's part of the `.available` check: 
![](https://screenshot.click/31-36-jedgn-llxgj.png)

Reply:
Yeah, +1 to do this as a follow-up. I think if we want to align quick add/quick order list with the buy button logic, we should look at both options (buy button using what QOL has) and consider what else we might need to add to both places, etc.

Logged an issue here: https://github.com/Shopify/dawn/issues/3499